public/app/features/trails/AutomaticMetricQueries/query-generators/general/index.test.ts

```diff
@@ -0,0 +1,40 @@
+import { SUFFIX_TO_EXPECTED_PARAMS } from '../__test__/suffix.cases';
+import { forEachKeyValue } from '../__test__/util';
+
+import * as queries from './queries';
+import * as rules from './rules';
+
+const generateQueries = jest.spyOn(queries, 'generateQueries');
+const getGeneratorParameters = jest.spyOn(rules, 'getGeneratorParameters');
+
+import general from '.';
+
+describe('general.generator', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('when metric has a "_sum" suffix', () => {
+    const metricParts = ['metric', 'with', 'suffix', 'sum'];
+    test('results in thrown error', () => {
+      expect(() => general.generator(metricParts)).toThrowError();
+    });
+  });
+
+  forEachKeyValue(SUFFIX_TO_EXPECTED_PARAMS, (suffix, expectedParams) => {
```

Comment:
instead of the custom forEachKeyValue util you could do:

```ts
for (const [suffix, expectedParams] of Object.entries(SUFFIX_TO_EXPECTED_PARAMS)) {
```

or import forEach from lodash.